pkg/common/utils.go

```diff
@@ -479,12 +474,49 @@ func existingErrorCode(err error) (codes.Code, error) {
 	if err == nil {
 		return codes.Unknown, fmt.Errorf("null error")
 	}
-	if status, ok := status.FromError(err); ok {
-		return status.Code(), nil
+	var tmpError *TemporaryError
+	// This explicitly checks our error is a temporary error before extracting its
+	// status, as there can be other errors that can qualify as statusable
+	// while not necessarily being temporary.
+	if errors.As(err, &tmpError) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
+	}
+	// We want to make sure we catch other temporary error types that are statusable.
+	// (eg. grpc-go/internal/status/status.go Error struct that wraps a status)
+	var googleErr *googleapi.Error
+	if !errors.As(err, &googleErr) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
 	}
+
 	return codes.Unknown, fmt.Errorf("no existing error code for %w", err)
 }
 
+func isAPIError(err error) (codes.Code, error) {
```

Comment:
I think isGoogleAPIError is probably a better name. 

Can you also add a function comment: 
isGoogleAPIError returns the gRPC status code for the given googleapi error by mapping the googleapi error's HTTP code to the corresponding gRPC error code.  If the error is wrapped in an APIError (github.com/googleapis/gax-go/v2/apierror), it maps the wrapped googleAPI error's HTTP code to the corresponding gRPC error code. Returns an error if the given error is not a googleapi error.